### PR TITLE
feat: config supports toggling HTTP and Stats servers on/off (#594)

### PR DESCRIPTION
**Pull request by jaw**
This changes the behavior of the stats server and git server but I assume most people don't want that by default.
The ssh/http servers are most likely the primary ones?

### CODE REVIEW

File: pkg/config/config.go [hunk 1]
```diff
@@ -57,6 +60,9 @@ type GitConfig struct {
 
 // HTTPConfig is the HTTP configuration for the server.
 type HTTPConfig struct {
+	// HttpServerEnabled toggles the HTTP server on/off
+	HttpServerEnabled bool `env:"HTTP_SERVER_ENABLED" yaml:"http_server_enabled"`
```
Comment (aymanbagabas) [inline pkg/config/config.go:64]:
```suggestion
	// Enabled toggles the HTTP server on/off
	Enabled bool `env:"HTTP_SERVER_ENABLED" yaml:"http_enabled"`
```

File: pkg/config/config.go [hunk 2]
```diff
@@ -39,6 +39,9 @@ type SSHConfig struct {
 
 // GitConfig is the Git daemon configuration for the server.
 type GitConfig struct {
+	// GitServerEnabled toggles the Git daemon on/off
+	GitServerEnabled bool `env:"GIT_SERVER_ENABLED" yaml:"git_server_enabled"`
```
Comment (aymanbagabas) [inline pkg/config/config.go:43]:
```suggestion
	// Enabled toggles the Git daemon on/off
	Enabled bool `env:"GIT_SERVER_ENABLED" yaml:"git_enabled"`
```

File: pkg/config/config.go [hunk 3]
```diff
@@ -72,6 +78,9 @@ type HTTPConfig struct {
 
 // StatsConfig is the configuration for the stats server.
 type StatsConfig struct {
+	// StatsServerEnabled toggles the HTTP server on/off
+	StatsServerEnabled bool `env:"STATS_SERVER_ENABLED" yaml:"stats_server_enabled"`
```
Comment (aymanbagabas) [inline pkg/config/config.go:82]:
```suggestion
	// Enabled toggles the HTTP server on/off
	Enabled bool `env:"STATS_SERVER_ENABLED" yaml:"stats_enabled"`
```

**Review by aymanbagabas — COMMENTED** [review]
This looks good from at a first glance. We need to be able to disable/enable all of them including http/ssh. We shouldn't assume people won't use them, especially since this is the default behavior right now. So let's make them enabled by default.

File: pkg/config/config.go (hunk 1 above)
Comment (jaw) [reply pkg/config/config.go]: Resolved

**Comment by jaw** [comment]
> LGTM! Finally, could you add tests to check if the servers are enabled/disabled

Not exactly sure what you're looking for...
Do you mean a test case where the env vars are verified to be working?

**Comment by aymanbagabas** [comment]
> Do you mean a test case where the env vars are verified to be working?

Yes. You could use `testscript`s to write the tests

**Comment by jaw** [comment]
> > Do you mean a test case where the env vars are verified to be working?
> 
> Yes. You could use `testscript`s to write the tests

Ack, done.

Tests got a little bit of polish as well.

**Comment by aymanbagabas** [comment]
> Tests got a little bit of polish as well.

Perfect! Could you put the test changes in a separate PR so that it doesn't get mixed up with the changes here?

**Comment by jaw** [comment]
This was rebased and submitted in another PR.

**Pull request closed without merging**